views/js/component/history/historyTable.js

```diff
@@ -0,0 +1,158 @@
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016 (original work) Open Assessment Technologies SA ;
+ */
+/**
+ * @author Jean-Sébastien Conan <jean-sebastien.conan@vesperiagroup.com>
+ */
+define([
+    'jquery',
+    'lodash',
+    'i18n',
+    'layout/loading-bar',
+    'ui/component',
+    'ui/datatable'
+], function ($, _, __, loadingBar, component) {
+    'use strict';
+
+    /**
+     * Some default config
+     * @type {Object}
+     * @private
+     */
+    var _defaults = {
+        sortBy: 'timestamp',
+        sortOrder: 'desc'
+    };
+
+    /**
+     * Renders the event details
+     * @param {String|Array} details
+     * @returns {String}
+     */
+    function renderDetails(details) {
+        if (_.isArray(details)) {
+            details = details.join('<br />');
+        }
+        return details;
+    }
+
+    /**
+     * Renders a history table
+     * @param {Object} config
+     * @param {String} config.service - The URL of the service providing the data
+     * @param {Object} [config.params] - A list of additional parameters to provide to the service
+     * @param {String} [config.sortBy] - The default sorted column
+     * @param {String} [config.sortOrder] - The default sort direction
+     * @param {Array} [config.tools] - A list of optional tools, using the datable format
+     * @param {Object} [data] - The first data set
+     * @returns {*}
+     */
+    function historyTableFactory(config, data) {
+        var initConfig = _.defaults(config || {}, _defaults);
+
+        // define some additional behavior
+        var historyTable = {
+            /**
+             * Refresh the table
+             * @param {Object} [params]
+             */
+            refresh: function refresh(params) {
+                var $element;
+
+                if (this.is('rendered')) {
+                    $element = this.getElement();
+                    if (params) {
+                        $element.datatable('options', {
+                            params: params
+                        });
+                    }
+                    $element.datatable('refresh');
+                } else {
+                    if (params) {
+                        initConfig.params = _.merge(initConfig.params, params);
+                    }
+                }
+            }
+        };
+
+        return component(historyTable)
+            .on('render', function() {
+                var self = this;
+                var tools = [{
+                    id: 'refresh',
+                    icon: 'reset',
+                    title: __('Refresh the page'),
+                    label: __('Refresh'),
+                    action: function() {
+                        self.refresh();
+                    }
+                }];
+                this.getElement()
+                    .on('query.datatable', function() {
+                        loadingBar.start();
```

Comment:
I wouldn't handle the loading bar from within the component but from the controller: trigger events, like `loading` and `loaded` and the controller will handle the loading bar, in order to decouple components from their context.